git/types.py

```diff
@@ -8,42 +8,39 @@
 from typing import (
     Dict,
     NoReturn,
-    Sequence,
+    Sequence as Sequence,
     Tuple,
     Union,
     Any,
+    Optional,
+    Callable,
     TYPE_CHECKING,
     TypeVar,
 )  # noqa: F401
 
-if sys.version_info[:2] >= (3, 8):
+if sys.version_info >= (3, 8):
     from typing import (
         Literal,
-        SupportsIndex,
         TypedDict,
         Protocol,
+        SupportsIndex as SupportsIndex,
         runtime_checkable,
     )  # noqa: F401
 else:
     from typing_extensions import (
         Literal,
-        SupportsIndex,  # noqa: F401
+        SupportsIndex as SupportsIndex,
         TypedDict,
         Protocol,
         runtime_checkable,
     )  # noqa: F401
 
-# if sys.version_info[:2] >= (3, 10):
+# if sys.version_info >= (3, 10):
```

Comment:
Is this a comment that is relevant to the typing system? If not, I think it can be deleted.

Reply:
Yes, relevant for some linting tools, like Pylance or mypy it seems.

Pylance most especially, will import packages and look for any `if-else` import, but only work if not manually reducing the tuple, you can easily notice as the text become grayed.

I think this change will also fix one of my issue I had with generating stubs with mypy.